microprofile-jwt/src/test/resources/pki/gen-keys.sh

```diff
@@ -1,15 +1,63 @@
 function gen_key_pair() {
 
-  PREFIX="${1}"
-  KEY_SIZE="${2}"
+  ALGORITHM="${1}"
+  PREFIX="${2}"
+  KEY_SIZE="${3}"
 
-  openssl genpkey -out "${PREFIX}.private.pkcs8.pem" -algorithm RSA -pkeyopt "rsa_keygen_bits:${KEY_SIZE}"
-  openssl rsa -in "${PREFIX}.private.pkcs8.pem" -pubout -outform PEM -out "${PREFIX}.public.pem"
+  if [[ "${ALGORITHM}" == "RS256" ]]; then
+    # RS256: RSASSA-PKCS1-v1_5 using SHA-256
+    mkdir -p ${ALGORITHM}
+    pushd ${ALGORITHM}
 
+    openssl genpkey -out "${PREFIX}.private.pkcs8.pem" -algorithm RSA -pkeyopt "rsa_keygen_bits:${KEY_SIZE}"
+    openssl rsa -in "${PREFIX}.private.pkcs8.pem" -pubout -outform PEM -out "${PREFIX}.public.pem"
+
+    popd
+  elif [[ "${ALGORITHM}" == "ES256" ]]; then
```

Comment:
Maybe just a styling issue, but maybe this can be split to three functions instead since the amount of shared code is low.

Reply:
@honza-kasik good point .... leaving a comment